compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -409,7 +403,13 @@ object Erasure extends TypeTestsCasts{
         }
       }
 
-      recur(typed(tree.qualifier, AnySelectionProto))
+      if (defn.DottyTupleNModule contains tree.qualifier.symbol) {
```

Comment:
What about making a `Set` version of `DottyTupleNModule`? This test is also used a few times in TupleRewrites.